src/main/java/org.wso2.apim.monetization/impl/workflow/StripeSubscriptionCreationWorkflowExecutor.java

```diff
@@ -0,0 +1,419 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.apim.monetization.impl.workflow;
+
+import com.google.gson.Gson;
+import com.stripe.Stripe;
+import com.stripe.exception.StripeException;
+import com.stripe.model.Customer;
+import com.stripe.model.Subscription;
+import com.stripe.model.Token;
+import com.stripe.net.RequestOptions;
+import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.wso2.apim.monetization.impl.StripeMonetizationConstants;
+import org.wso2.apim.monetization.impl.StripeMonetizationDAO;
+import org.wso2.apim.monetization.impl.StripeMonetizationException;
+import org.wso2.apim.monetization.impl.model.MonetizationPlatformCustomer;
+import org.wso2.apim.monetization.impl.model.MonetizationSharedCustomer;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.api.WorkflowResponse;
+import org.wso2.carbon.apimgt.api.model.API;
+import org.wso2.carbon.apimgt.api.model.APIIdentifier;
+import org.wso2.carbon.apimgt.api.model.Subscriber;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.dao.ApiMgtDAO;
+import org.wso2.carbon.apimgt.impl.dto.SubscriptionWorkflowDTO;
+import org.wso2.carbon.apimgt.impl.dto.WorkflowDTO;
+import org.wso2.carbon.apimgt.impl.internal.ServiceReferenceHolder;
+import org.wso2.carbon.apimgt.impl.workflow.GeneralWorkflowResponse;
+import org.wso2.carbon.apimgt.impl.workflow.WorkflowConstants;
+import org.wso2.carbon.apimgt.impl.workflow.WorkflowException;
+import org.wso2.carbon.apimgt.impl.workflow.WorkflowExecutor;
+import org.wso2.carbon.apimgt.impl.workflow.WorkflowStatus;
+import org.wso2.carbon.registry.core.Registry;
+import org.wso2.carbon.registry.core.Resource;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+
+import java.nio.charset.Charset;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * worrkflow executor for stripe based subscription create action
+ */
+public class StripeSubscriptionCreationWorkflowExecutor extends WorkflowExecutor {
+
+    private static final Log log = LogFactory.getLog(StripeSubscriptionCreationWorkflowExecutor.class);
+    StripeMonetizationDAO stripeMonetizationDAO = StripeMonetizationDAO.getInstance();
+
+    @Override
+    public String getWorkflowType() {
+
+        return WorkflowConstants.WF_TYPE_AM_SUBSCRIPTION_CREATION;
+    }
+
+    @Override
+    public List<WorkflowDTO> getWorkflowDetails(String workflowStatus) throws WorkflowException {
+
+        return null;
+    }
+
```

Comment:
Please remove unwanted new lines in each file.

Reply:
fixed in 49ded003cbdb2832bd0e8845951ac7af9831f3cb